### PR TITLE
Move Provisioning API to the AppFramework

### PR DESCRIPTION
**Pull request by rullzer**
Requires: https://github.com/nextcloud/server/pull/844 and https://github.com/nextcloud/server/pull/845

Lets see what CI tells us

### CODE REVIEW

**Comment by rullzer** [comment]
Wheehee intergration tests are happy!

Review time: @LukasReschke @MorrisJobke @nickvergessen @schiessle @icewind1991

File: apps/provisioning_api/lib/Controller/GroupsController.php [hunk 1]
```diff
@@ -114,59 +119,62 @@ public function getGroup($parameters) {
...
+			throw new OCSException('User does not have access to specified group', \OCP\API::RESPOND_UNAUTHORISED);
 		}
 	}
 
 	/**
 	 * creates a new group
 	 *
-	 * @param array $parameters
-	 * @return \OC\OCS\Result
+	 * @NoAdminRequired
```
Comment (LukasReschke) [inline apps/provisioning_api/lib/Controller/GroupsController.php]: Is there any middleware in place to require at least subadmin rights?

Comment (LukasReschke) [inline apps/provisioning_api/lib/Controller/GroupsController.php]: https://github.com/nextcloud/server/blob/da64baa378ae32c41e653155cc990c3046158e56/apps/provisioning_api/lib/Middleware/ProvisioningApiMiddleware.php seems so. Huuuuge PR 🙈

File: apps/provisioning_api/lib/Controller/UsersController.php [hunk 2]
```diff
@@ -106,89 +116,86 @@ public function getUsers() {
...
-	 * @return \OC\OCS\Result
+	 * @param string $userId
+	 * @return DataResponse
+	 * @throws OCSException
 	 */
-	public function getUser($parameters) {
-		$userId = $parameters['userid'];
-
+	public function getUser($userId) {
 		// Check if user is logged in
```
Comment (LukasReschke) [inline apps/provisioning_api/lib/Controller/UsersController.php]: That comments is kinda confusing now, kill it with 🔥 ?

Comment (LukasReschke) [inline apps/provisioning_api/lib/Controller/UsersController.php]: It's also at a few other places in the code :)

File: apps/provisioning_api/lib/Controller/GroupsController.php (hunk 1 above)
Comment (rullzer) [inline apps/provisioning_api/lib/Controller/GroupsController.php]: Yes :)

File: apps/provisioning_api/lib/Controller/UsersController.php (hunk 2 above)
Comment (rullzer) [inline apps/provisioning_api/lib/Controller/UsersController.php]: Ah I removed the code but not the comments apparently...
Will delete...

Comment (rullzer) [inline apps/provisioning_api/lib/Controller/UsersController.php]: g0ne

**Comment by LukasReschke** [comment]
LGTM

**Comment by nickvergessen** [comment]
:+1:

**Pull request merged**